dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/netty4/h2/NettyHttp2ProtocolSelectorHandler.java

```diff
@@ -68,8 +69,9 @@
             h2StreamChannel = new Http2WriteQueueChannel(h2StreamChannel, writeQueue);
         }
         ChannelPipeline pipeline = ctx.pipeline();
-        pipeline.addLast(
-                new NettyHttp2FrameHandler(h2StreamChannel, factory.newInstance(h2StreamChannel, url, frameworkModel)));
+        Http2TransportListener http2TransportListener = factory.newInstance(h2StreamChannel, url, frameworkModel);
+        ctx.channel().closeFuture().addListener(future -> http2TransportListener.onStreamClosed());
```

Comment:
**Security Issue**: Automatically closing the stream on receiving any close future event without validating the cause or source of the close request may inadvertently allow an attacker to trigger unintended closures, potentially leading to denial of service. <br> **Fix**: Validate the cause of the close future event to ensure it's legitimate and expected. Implement logic to differentiate between normal and abnormal closures, possibly inspecting the close status or reason if available. <br> **Code Suggestion**: 
 ```
 if (validateCloseEvent(closeFutureEvent)) {
     stream.close();
 }
 ```